src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -656,102 +577,71 @@ public void startSession(String path, String userName, String botName) throws IO
    * 
    * @param path
    *          - he path to the ProgramAB directory where the bots aiml resides
+   *          FIXME - path is not needed
    * @param userName
    *          - The new user name
    * @param botName
    *          - The name of the bot to load. (example: alice2)
    * @param locale
    *          - The locale of the bot to ensure the aiml is loaded (mostly for
-   *          Japanese support)
-   * @throws IOException 
+   *          Japanese support) FIXME - local is defined in the bot,
+   *          specifically config/mrl.properties
+   * @throws IOException
    */
-  public void startSession(String path, String userName, String botName, java.util.Locale locale) throws IOException {
-    // if update the current user/bot name globally. (bring this bot/user
-    // session to attention.)
-    log.info("Start Session Path: {} User: {} Bot: {} Locale: {}", path, userName, botName, locale);
-    
-    File check = new File(path + fs + "bots" + fs + botName);
-    if (!check.exists() || !check.isDirectory()) {
-      String invalid = String.format("%s could not load aiml. %s is not a valid bot directory", getName(), check.getAbsolutePath());
-      error(invalid);
-      throw new IOException(invalid);
-    }
-    
-    updateCurrentSession(userName, botName);
-    // Session is between a user and a bot. key is compound.
-    if (sessions.containsKey(botName) && sessions.get(botName).containsKey(userName)) {
-      info("Session %s %s already created", botName, userName);
-      return;
-    }
-    setReady(false);
-    info("Starting chat session path: %s username: %s botname: %s", path, userName, botName);
-    setPath(path);
-    setCurrentBotName(botName);
-    setCurrentUserName(userName);
-    // check if we've already started this bot.
-    Bot bot = bots.get(botName.toLowerCase());
-    if (bot == null) {
-      // create a new bot if we haven't started it yet.
-      bot = new Bot(botName, path, locale);
-      // Hijack all the SRAIX requests and implement them as a synchronous call
-      // to
-      // a service to return a string response for programab...
-      bot.setSraixHandler(new MrlSraixHandler(this));
-      // put the bot into the bots map/cache referenced by it's lowercase
-      // botName.
-      bots.put(botName.toLowerCase(), bot);
+
+  @Deprecated /*

```

Comment:
you've marked this as deprecated, but I don't see a new version of startSession to use instead.  You should only mark something deprecated if there is a replacement way of doing it.

Reply:
replacement would be startSession(String userName, String botName) which already exists - I'll comment it in ..

Reply:
Then this code should be removed and call the other method.  however, doing so, you will loose functionality of specifying the locale.  This is probably ok.  If this is the way we are going forward, this method should be cleaned out, leaving only a warning message and a call to the proper method...  it means that the locale parameter will be ignored.

Reply:
added comment /* use startSession(String userName, String botName) */